src/wp-includes/media.php

```diff
@@ -5614,3 +5832,45 @@ function wp_increase_content_media_count( $amount = 1 ) {
 
 	return $content_media_count;
 }
+
+/**
+ * Add fetchpriority='high' to $loading_attrs if applicable.
+ *
+ * @since 6.3.0
+ * @access private
+ *
+ * @param array[] $loading_attrs Array of the loading attributes for the element.
+ * @param array[] $attr Array of the attributes for the element.
+ * @return array[] $loading_attrs Updated loading attributes for the element.
+ */
+function wp_maybe_add_fetchpriority_high_attr( $loading_attrs, $attr ) {
```

Comment:
@felixarntz  I have not added lazy-loading here as i felt this check should be at the start of `wp_get_loading_optimization_attributes` to avoid additional checks.

Reply:
See my comment above, I think we should protect this method against that so that it's addressed at the root. This function takes the responsibility of conditionally adding `fetchpriority="high"`, so if there is already `loading="lazy"`, we should ensure `fetchpriority="high"` is not added.

Doing that check is not expensive, so even if it's a duplicate, it's important that it's implemented here so we prevent a scenario where both attributes are possibly used together because of WP core.